### PR TITLE
Bug 1817726 - Add Recents url sharing

### PR DESCRIPTION
**Pull request by rahulsainani**
Enables [Recents URL sharing](https://developer.android.com/guide/components/activities/recents#url-sharing) so users can share links to recently viewed web content directly from the Recents screen.

### Video
https://user-images.githubusercontent.com/38040960/220072399-6fafcdfa-d6c3-4d68-b11d-3c603c249ab6.mp4

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1817726

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 1]
```diff
@@ -523,6 +526,14 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         BrowsersCache.resetAll()
     }
 
+    override fun onProvideAssistContent(outContent: AssistContent?) {
+        super.onProvideAssistContent(outContent)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (Mugurell) [inline fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:531]: The documentation mentions "On Pixel devices running Android 12 or higher" so I'm curious why are we using a lower api level check here?
Based on the fact that this method [was added in API 23](https://developer.android.com/reference/android/app/Activity#onProvideAssistContent(android.app.assist.AssistContent)) and the official examples[1][2] not doing an api check maybe we can avoid it also?



[1] https://developer.android.com/guide/components/activities/recents#url-sharing
[2] https://developer.android.com/training/articles/assistant.html#source_app

**Review by Mugurell — APPROVED** [review]
Nice!
This seems to automatically consider `FLAG_SECURE` and not show the url options while in private mode which is great!

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt (hunk 1 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt]: Yes you're right! Thanks for sharing the assistant link as well, TIL! 👌 I updated to remove the if check and added the annotation to suppress the lint issue as `setWebUrl` function along the function was added in api 23.

**Comment by rahulsainani** [comment]
@Mergifyio backport releases_v111

**Pull request merged**